### DISCUSSION TITLE
Explicit many-to-many relationship updating the relation table is not working

### DISCUSSION THREAD

**Discussion by gurjotkaur20** [prisma/prisma#23131] [Q&A] [answered]
### Bug description

Unable to update the relation table as implicit many to many relations.

### How to reproduce

Below is the prisma schema:
```prisma
model user {
  id    String     @id @default(uuid())
  email String  @unique
  groups users_groups[]
}

model groupp {
  id String @id @default(uuid())
  name String
  users users_groups[]
}

model users_groups {
  user     user    @relation(fields: [userId], references: [id])
  userId   String
  group      groupp     @relation(fields: [groupId], references: [id])
  groupId    String

  @@id([userId, groupId])
}
```

I am creating groupp as follows:
```js
await prisma.groupp.create({
    data: {
      name: 'group1',
      users: {
        create: {
          user: {
            connect: {
              id: 'bc535a2f-a833-46cb-9ad6-4c3fa1bc7378'
            }}}}}});
```

It creates one row in `users_groups` table. But when I try to update the groupp and wants to link to some other user using the below query then it again creates a new row in `users_groups` table.
```js
await prisma.groupp.update({
    where: {
      id: 'b81f8b10-9bf6-4589-b95b-fe55430a47f5'
    },
    data: {
      users: { create: { user: { connect: { id: userId } } } },
    },
  });
```

But now it will give me two users attach to this groupp. How can I overwrite the user and groupp relation and just link one new user to the groupp?

### Expected behavior

When I am using implicit many to many relations then just using set, I am able to override the existing link and just link one user to the groupp. This is not working in explicit many to many relations. For example, below query in implicit many to many will remove the existing linked user and will just set new users.
```js
await prisma.groupp.update({
    where: {
      id: 'b81f8b10-9bf6-4589-b95b-fe55430a47f5'
    },
    data: {
      users: { set: [ { id: userId1 }, { id: userId2 }] },
    },
  });
```

### Prisma information

```prisma
model user {
  id    String     @id @default(uuid())
  email String  @unique
  groups users_groups[]
}

model groupp {
  id String @id @default(uuid())
  name String
  users users_groups[]
}

model users_groups {
  user     user    @relation(fields: [userId], references: [id])
  userId   String
  group      groupp     @relation(fields: [groupId], references: [id])
  groupId    String

  @@id([userId, groupId])
}
```

### Environment & setup

- OS: Debian
- Database: MySQL
- Node.js version: v16.14.2

### Prisma Version

```
"prisma": "^5.2.0"
```

**Comment by ludralph** [comment, marked as answer]
Hi @gurjotkaur20 👋 

Thank you for raising this question.

In the context of explicit many-to-many relationships, Prisma does not directly support the `set` operation as it does with implicit relationships. This is because, with explicit relationships, you are directly managing the join table (`users_groups` in your case), and Prisma expects you to handle the creation and deletion of records in this table yourself.
However, a workaround to achieve the behavior you're looking for—replacing existing relations with new ones—can be done in two steps:

- Delete the existing relations from the join table.
- Create new relations in the join table.

After clearing the existing relations, you then create new ones by updating the `groupp model` with a create operation under the users field. This operation adds new entries to the users_groups table, linking the groupp to new user(s).

```js
// Step 1: Delete existing relations for the group
await prisma.users_groups.deleteMany({
  where: {
    groupId: 'b81f8b10-9bf6-4589-b95b-fe55430a47f5',
  },
});

// Step 2: Update the group and create or connect new user relations
await prisma.groupp.update({
  where: {
    id: 'b81f8b10-9bf6-4589-b95b-fe55430a47f5',
  },
  data: {
    users: {
      create: [
        { user: { connect: { id: 'newUserId1' } } },
        // Add more users as needed
      ],
    },
  },
});
```
This approach requires two separate operations and might not be as efficient or atomic as a single set operation in implicit many-to-many relations. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by gurjotkaur20** [comment]
Thanks @ludralph

**Reply by ludralph** [reply]
You're welcome 👍
